scripts/generate_readme/cli.py

```diff
@@ -0,0 +1,150 @@
+    """Validate that the component directory exists and contains required files.
+    
+    Args:
+        dir_path: String path to the component directory.
+        
+    Returns:
+        Path: Validated Path object to the component directory.
+        
+    Raises:
+        argparse.ArgumentTypeError: If validation fails.
+    """
+    path = Path(dir_path)
+    
+    if not path.exists():
+        raise argparse.ArgumentTypeError(f"Component directory '{dir_path}' does not exist")
+    
+    if not path.is_dir():
+        raise argparse.ArgumentTypeError(f"'{dir_path}' is not a directory")
+    
+    component_file = path / 'component.py'
+    if not component_file.exists():
+        raise argparse.ArgumentTypeError(f"'{dir_path}' does not contain a component.py file")
+    
+    metadata_file = path / 'metadata.yaml'
+    if not metadata_file.exists():
+        raise argparse.ArgumentTypeError(f"'{dir_path}' does not contain a metadata.yaml file")
+    
+    return path
+
+
+def validate_pipeline_directory(dir_path: str) -> Path:
+    """Validate that the pipeline directory exists and contains required files.
+    
+    Args:
+        dir_path: String path to the pipeline directory.
+        
+    Returns:
+        Path: Validated Path object to the pipeline directory.
+        
+    Raises:
+        argparse.ArgumentTypeError: If validation fails.
+    """
+    path = Path(dir_path)
+    
+    if not path.exists():
+        raise argparse.ArgumentTypeError(f"Pipeline directory '{dir_path}' does not exist")
+    
+    if not path.is_dir():
+        raise argparse.ArgumentTypeError(f"'{dir_path}' is not a directory")
+    
+    pipeline_file = path / 'pipeline.py'
+    if not pipeline_file.exists():
+        raise argparse.ArgumentTypeError(f"'{dir_path}' does not contain a pipeline.py file")
+    
+    metadata_file = path / 'metadata.yaml'
+    if not metadata_file.exists():
+        raise argparse.ArgumentTypeError(f"'{dir_path}' does not contain a metadata.yaml file")
+    
+    return path
+
+
+def parse_arguments():
+    """Parse and validate command-line arguments.
+    
+    Returns:
+        argparse.Namespace: Parsed and validated arguments.
+    """
+    parser = argparse.ArgumentParser(
+        description="Generate README.md documentation for Kubeflow Pipelines components and pipelines",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        epilog="""
+Examples:
+  # From project root:
+  python -m scripts.generate_readme --component components/some_category/my_component
+  python -m scripts.generate_readme --pipeline pipelines/some_category/my_pipeline
+  python -m scripts.generate_readme --component components/some_category/my_component --output custom_readme.md
+  python -m scripts.generate_readme --component components/some_category/my_component --verbose --overwrite
+  
+  # Or with uv:
+  uv run python -m scripts.generate_readme --component components/some_category/my_component
+        """
+    )
+    
```

Comment:
@gmfrasca Could you also add the category-level README.md index generation as mentioned in the KEP-

`Category `README.md` files act as indices. Each lists the components/pipelines in that category, provides one-line
  summaries, and links to the corresponding asset directories; these files are generated automatically and kept fresh by
  the README automation described above.`